packages/portfolio-contract/src/portfolio.exo.ts

```diff
@@ -39,195 +45,187 @@ export const DECODE_CONTRACT_CALL_RESULT_ABI = [
       },
     ],
   },
-] as const;
-harden(DECODE_CONTRACT_CALL_RESULT_ABI);
-
-export const supportedEVMChains: CaipChainId[] = [
-  'eip155:43114', // Avalanche
-  'eip155:8453', // Base
-  'eip155:1', // Ethereum
-] as const;
-harden(supportedEVMChains);
+];
 
 const OrchestrationAccountShape = M.remotable('OrchestrationAccount');
 const VowStringShape = M.any(); // Vow(M.string())
 const KeeperI = M.interface('keeper', {
   getLCA: M.call().returns(OrchestrationAccountShape),
   getPositions: M.call().returns(M.arrayOf(M.string())),
-  initAave: M.call(ChainShape).returns(),
-  initCompound: M.call().returns(),
-  initUSDN: M.call(OrchestrationAccountShape).returns(),
   getUSDNICA: M.call().returns(OrchestrationAccountShape),
-  getAaveAddress: M.call().returns(VowStringShape),
-  getCompoundAddress: M.call().returns(VowStringShape),
+  initAave: M.call(M.string()).returns(),
+  initCompound: M.call(M.string()).returns(),
+  initUSDN: M.call(M.remotable('OrchestrationAccount')).returns(),
+  initGmp: M.call(M.remotable('OrchestrationAccount')).returns(),
+  getRemoteAddress: M.call().returns(VowStringShape),
+  wait: M.call(M.bigint()).returns(VowShape),
 });
 
-const EvmTapI = M.interface('EvmTap', {
-  receiveUpcall: M.call(M.record()).returns(M.promise()),
-});
+type AxelarGmpManager = {
+  localAccount: LocalAccount;
+  remoteAddressVK: VowKit<`0x${string}`>;
+};
 
 type EVMProtocolState = {
   chain: CaipChainId;
-  remoteAddressVK: VowKit<`0x${string}`>;
 };
 
-const EthL1: CaipChainId = 'eip155:1';
-
-/** ICA on Noble */
-type NobleAccount = OrchestrationAccount<{ chainId: 'noble-any' }>;
-
 type PortfolioKitState = {
-  localAccount: LocalAccount;
   Aave: EVMProtocolState | undefined;
   Compound: EVMProtocolState | undefined;
   USDN: NobleAccount | undefined;
+  Gmp: AxelarGmpManager | undefined;
```

Comment:
as I said above, I suggest inlining this...

```suggestion
  Gmp: undefined | {
    localAccount: HostInterface<LocalAccount>;
    remoteAddressVK: VowKit<`0x${string}`>;
  };
```